### PR TITLE
Add Support for SSO.

### LINKED ISSUE #335: Add support for SSO (social login)
Similar to Open Liberty images, add support for single sign-on (SSO) using social logins.

- Add configuration snippets and scripts to configure one or more social login providers
- Add documentation

### PR DESCRIPTION
**Pull request by InfTkm**
Changed configure.sh files in the latest and 20.0.0.5 releases.
SECURITY.md updated according to OL.
Copied SSO related XML files from OL.

Fixes #335.

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -20,8 +20,96 @@ Container also can import certificates from Kubernetes.
...
 
 
-### Providing custom keystore
+### Providing a custom keystore
 
 A custom keystore can be provided during the application image's build phase by simply copying the keystore into the image's  `/output/resources/security/key.p12` location. 
 
 You must then override the keystore's password by including your copy of the `keystore.xml` file inside the `/config/configDropins/defaults/` directory.
+
+## Single Sign-On configuration
```
Comment (leochr) [inline SECURITY.md:29]: Please add (`20.0.0.5`) to indicate when this support was added to WebSphere Liberty images.

Also, enhance the description for the Security section in the main page: similar to this for OL - https://github.com/OpenLiberty/ci.docker#security

File: SECURITY.md [hunk 2]
```diff
@@ -20,8 +20,96 @@ Container also can import certificates from Kubernetes.
...
+  * They can be passed as environment variables to the Docker container when it is deployed. 
+  * They can be supplied in a deployment YAML file or by the [Liberty operator](https://github.com/OpenLiberty/open-liberty-operator/blob/master/doc/user-guide.adoc#single-sign-on-sso), which will pass them to the container at deploy time.
+
+Client ID and Client Secret are obtained from the provider.  RedirectToRPHostAndPort (`SEC_SSO_REDIRECTTORPHOSTANDPORT`) is the protocol, host, and port that the provider should send the browser back to after authentication, for example `https://myApp-myNamespace-myClusterHostname.mycompany.com`  (In some container environments, the pod cannot figure this out and it will need to be specified.) Other variables may be needed in some situations and are documented in detail in the [Open Liberty Documentation](https://openliberty.io/docs/ref/feature/#socialLogin-1.0.html) under each type of provider. The `oidc` and `oauth2` configurations are general purpose configurations for use with any provider that uses the OpenID Connect 1.0 or OAuth 2.0 specifications.
+
+A sample Dockerfile and Liberty operator YAML file are [here](samples/security).
```
Comment (leochr) [inline SECURITY.md:56]: Please add the sample artifacts to WL repo (similar to OL but change the base image)

File: ga/20.0.0.5/kernel/helpers/build/configure.sh [hunk 3]
```diff
@@ -5,111 +5,156 @@ fi
...
+  find /opt/ibm/wlp -perm -g=w -print0 | xargs -0 -r chmod -R g+rw
+  # Create a new SCC layer
+  if [ "$OPENJ9_SCC" == "true" ]
+  then
+    populate_scc.sh
+  fi
+  #Make folder executable for a group
+  find /opt/ibm/wlp -type d -perm -g=x -print0 | xargs -0 -r chmod -R g+rwx
+
+  if [[ -n "$SEC_SSO_PROVIDERS" ]]; then
```
Comment (leochr) [inline ga/20.0.0.5/kernel/helpers/build/configure.sh:118]: This should be added prior to installing necessary features (line with # Install needed features), so the features being enabled by these snippets can be installed by installUtility.

**Comment by leochr** [comment]
Thanks for the PR @InfTkm. Please see my comments above.

File: SECURITY.md (hunk 1 above)
Comment (InfTkm) [reply SECURITY.md]: There is another subsection called **Security** under top level section **OpenJ9 Shared Class Cache (SCC)**. I added a top level section called **Security** same as the OL repo. Hopefully it's not confusing.

File: SECURITY.md (hunk 2 above)
Comment (InfTkm) [reply SECURITY.md:56]: The artifacts are added. I am not sure if I should make any changes to the `olapp.yaml` file though.

Comment (leochr) [reply SECURITY.md:56]: Thanks. `olapp.yaml` can be the same as what we have in OL repo.

File: README.md [hunk 4]
```diff
@@ -70,6 +70,11 @@ This section describes the optional enterprise functionality that can be enabled
 *  `VERBOSE`
   * Description: When set to `true` it outputs the commands and results to stdout from `configure.sh`. Otherwise, default setting is `false` and `configure.sh` is silenced.
 
+## Security
```
Comment (leochr) [inline README.md:73]: We can keep this section and get rid of the `Security` section below. 

Thanks for fixing some broken links.

Unrelated to this PR, but will be good to fix these:
- The headings, starting with `Logging` and until 
`Installing Liberty Features from local repository (19.0.0.8+)` are not right. They shouldn't be under `OpenJ9 Shared Class Cache (SCC)`. Please move them one level up (### --> ##). 
- Spell checks: replace all instances of `Decription` with `Description` and replace `Locallly` with `Locally`

**Pull request merged**